### PR TITLE
Fetch current detail controller and compare

### LINKED ISSUE #244: Selecting the currently selected issue reloads the detail view.
Very low priority tbh just adding for posterity because it’s a less than ideal UX

Version 1.5.0 (1330)
Device: iPad7,3 (iOS 11.0)

### PR DESCRIPTION
**Pull request by rob-nash**
This pull request prevents the detail view from being reloaded unnecessarily. Which closes the following.

Closes #244

### CODE REVIEW

File: Classes/Notifications/NotificationSectionController.swift [hunk 1]
```diff
@@ -97,6 +101,10 @@ final class NotificationSectionController: ListSwiftSectionController<Notificati
             showRelease(release, model: model)
         }
     }
+    
+    private var currentDetailController: IssuesViewController? {
+        return (UIApplication.shared.delegate as? AppDelegate)?.rootNavigationManager.detailNavigationController?.topViewController as? IssuesViewController
```
Comment (jdisho) [inline Classes/Notifications/NotificationSectionController.swift]: Would be better to split this up

File: Classes/Issues/IssueDetailsModel.swift [hunk 2]
```diff
@@ -15,3 +15,5 @@ struct IssueDetailsModel {
     let number: Int
 
 }
+
+extension IssueDetailsModel: Equatable {}
```
[the remark this answers is not in the document]
Comment (rob-nash) [reply Classes/Issues/IssueDetailsModel.swift]: Sure

File: Classes/Issues/IssuesViewController.swift [hunk 3]
```diff
@@ -36,7 +36,7 @@ final class IssuesViewController:
     MessageTextViewListener {
 
     private let client: GithubClient
-    private let model: IssueDetailsModel
+    let model: IssueDetailsModel
```
Comment (jdisho) [inline Classes/Issues/IssuesViewController.swift]: Why did you change the access control? If there is a reason, move it before the `private` properties 😊

File: Classes/Notifications/NotificationSectionController.swift [hunk 4]
```diff
@@ -86,9 +86,13 @@ final class NotificationSectionController: ListSwiftSectionController<Notificati
         case .hash(let hash):
             viewController?.presentCommit(owner: model.owner, repo: model.repo, hash: hash)
         case .number(let number):
+            let model = IssueDetailsModel(owner: model.owner, repo: model.repo, number: number)
+            if let controller = currentDetailController, controller.model == model {
+                return
+            }
             let controller = IssuesViewController(
```
Comment (jdisho) [inline Classes/Notifications/NotificationSectionController.swift]: Would be better to change the name to `issuesViewController`, because atm there are two controllers with the same name.

File: Classes/Notifications/NotificationSectionController.swift (hunk 1 above)
Comment (rob-nash) [reply Classes/Notifications/NotificationSectionController.swift]: Ok

File: Classes/Issues/IssuesViewController.swift (hunk 3 above)
Comment (rob-nash) [reply Classes/Issues/IssuesViewController.swift]: I did this so that I could reach the property from the `NotificationSectionController`

Comment (rob-nash) [reply Classes/Issues/IssuesViewController.swift]: Have moved

File: Classes/Notifications/NotificationSectionController.swift (hunk 4 above)
Comment (rob-nash) [reply Classes/Notifications/NotificationSectionController.swift]: Done
